src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -0,0 +1,149 @@
+    private static final String DEFAULT_STORAGE = "default";
+
+    private final SecuritySettings securitySettings;
+    private final EventTypeCache eventTypeCache;
+    private final StorageDbRepository storageDbRepository;
+    private final TimelineSync timelineSync;
+    private final NakadiSettings nakadiSettings;
+    private final TimelineDbRepository timelineDbRepository;
+    private final TopicRepositoryHolder topicRepositoryHolder;
+
+    @Autowired
+    public TimelineService(final SecuritySettings securitySettings,
+                           final EventTypeCache eventTypeCache,
+                           final StorageDbRepository storageDbRepository,
+                           final TimelineSync timelineSync,
+                           final NakadiSettings nakadiSettings,
+                           final TimelineDbRepository timelineDbRepository,
+                           final TopicRepositoryHolder topicRepositoryHolder) {
+        this.securitySettings = securitySettings;
+        this.eventTypeCache = eventTypeCache;
+        this.storageDbRepository = storageDbRepository;
+        this.timelineSync = timelineSync;
+        this.nakadiSettings = nakadiSettings;
+        this.timelineDbRepository = timelineDbRepository;
+        this.topicRepositoryHolder = topicRepositoryHolder;
+    }
+
+    public void createTimeline(final TimelineRequest timelineRequest, final Client client)
+            throws ForbiddenAccessException, TimelineException, TopicRepositoryException {
+        if (!client.getClientId().equals(securitySettings.getAdminClientId())) {
+            throw new ForbiddenAccessException("Request is forbidden for user " + client.getClientId());
+        }
+
+        try {
+            final String eventTypeName = timelineRequest.getEventType();
+            final EventType eventType = eventTypeCache.getEventType(eventTypeName);
+
+            final Storage storage = storageDbRepository.getStorage(timelineRequest.getStorageId())
+                    .orElseThrow(() -> new NotFoundException("No storage with id: " + timelineRequest.getStorageId()));
+            final Timeline activeTimeline = getTimeline(eventType);
+            final TopicRepository currentTopicRepo =
+                    topicRepositoryHolder.getTopicRepository(activeTimeline.getStorage());
+            final TopicRepository nextTopicRepo = topicRepositoryHolder.getTopicRepository(storage);
+            final List<PartitionStatistics> partitionStatistics =
+                    currentTopicRepo.loadTopicStatistics(Collections.singleton(activeTimeline.getTopic()));
+
+            if (activeTimeline.isFake()) {
+                LOG.info("Switching from fake timeline to real one. Current topic {}", activeTimeline.getTopic());
+                final Timeline nextTimeline = Timeline.createTimeline(activeTimeline.getEventType(),
+                        activeTimeline.getOrder() + 1, storage, activeTimeline.getTopic(), new Date());
```

Comment:
Ok, so it will switch from fake timeline to real timeline but the actual topic will stay the same, right? Why do we need this step? Can we go from fake timeline directly to real one with different topic?